controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -375,7 +381,14 @@ public String getScopeName() {
 
                         log.info("Scale {}/{} for segments started. Creating new segments. SegmentsToSeal {}", scope, name, sealedSegments);
                         // fresh run
-                        return scaleCreateNewSegments(newRanges, scaleTimestamp, segmentTable, activeEpoch);
+                        // Ensure that segment.creation time is monotonically increasing after each new scale
+                        long lastScaleTime = HistoryRecord.readLatestRecord(historyTable.getData(), true).map(HistoryRecord::getScaleTime).orElse(0L);
+
+                        long scaleEventTime = Math.max(System.currentTimeMillis(), scaleTimestamp);
+
+                        long segmentCreationTimestamp = Math.max(scaleEventTime, lastScaleTime + 1);
```

Comment:
I think the logic we trying to implement here is that `segmentCreationTimestamp` should be `lastScaleTime + 1` in the case the `scaleTimestamp` is smaller than `lastScaleTime`.

Reply:
replied above